datafusion/expr/src/built_in_function.rs

```diff
@@ -315,6 +320,72 @@ fn function_to_name() -> &'static HashMap<BuiltinScalarFunction, &'static str> {
     })
 }
 
+/// Returns the wider type among lhs and rhs.
+/// Wider type is the type that can safely represent the other type without information loss.
+/// Returns Error if types are incompatible.
+fn get_wider_type(lhs: &DataType, rhs: &DataType) -> Result<DataType> {
```

Comment:
Would it make sense to move this function into the `expr::type_coercion` module, where we have similar functions, such as `get_wider_decimal_type`?

Reply:
This makes sense, I moved the function under `type_coercion` module. Thanks!.